src/ewatercycle/observation/grdc.py

```
@@ -300,71 +300,50 @@ def _grdc_metadata_reader(grdc_station_path, all_lines):  # noqa: C901
+            all_lines, "Owner of original data", default="Unknown"
+        )
 
-        try:
-            attribute_grdc["grdc_catchment_area_in_km2"] = float(
-                all_lines[14].split(":")[1].strip()
-            )
-            if attribute_grdc["grdc_catchment_area_in_km2"] <= 0.0:
-                attribute_grdc["grdc_catchment_area_in_km2"] = "NA"
-        except (IndexError, ValueError):
+        if (
+            attribute_grdc["grdc_catchment_area_in_km2"] != "NA"
+            and attribute_grdc["grdc_catchment_area_in_km2"] <= 0.0
+        ):
             attribute_grdc["grdc_catchment_area_in_km2"] = "NA"
 
-        try:
-            attribute_grdc["altitude_masl"] = float(all_lines[15].split(":")[1].strip())
-        except (IndexError, ValueError):
-            attribute_grdc["altitude_masl"] = "NA"
-
-        try:
-            attribute_grdc["dataSetContent"] = str(all_lines[20].split(":")[1].strip())
-        except (IndexError, ValueError):
-            attribute_grdc["dataSetContent"] = "NA"
-
-        try:
-            attribute_grdc["units"] = str(all_lines[22].split(":")[1].strip())
-        except (IndexError, ValueError):
-            attribute_grdc["units"] = "NA"
+    return attribute_grdc
 
-        try:
-            attribute_grdc["Owner of original data"] = (
-                all_lines[18].split(":")[1].strip()
-            )
-        except (IndexError, ValueError):
-            attribute_grdc["Owner of original data"] = "Unknown"
 
-    return attribute_grdc
+def _extract_metadata(lines, key, cast=str, default="NA"):
+    """Private helper to extract metadata fields from GRDC header lines."""
+    for line in lines:
+        if line.strip().startswith(f"# {key}:"):
+            parts = line.split(":", 1)
+            if len(parts) > 1:
+                value = parts[1].strip()
+                try:
+                    return cast(value)
+                except ValueError:
+                    return default
```

I think this can lead to unexpected behavior. Might be better to raise an error with a clear message here (or use python's default error if that one is clear).